sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppImpl.java

```diff
@@ -360,4 +369,47 @@ public Mono<KuduDeploymentResult> pushDeployAsync(DeployType type, File file, De
             return Mono.error(e);
         }
     }
+
+    @Override
+    public DeploymentBuildStatus getDeploymentStatus(String deploymentId) {
+        return getDeploymentStatusAsync(deploymentId).block();
+    }
+
+    @Override
+    public Mono<DeploymentBuildStatus> getDeploymentStatusAsync(String deploymentId) {
+        // "GET" LRO is not supported in azure-core
+        String deploymentStatusUrl = String.format(
+            "%s/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Web/sites/%s/deploymentStatus/%s?api-version=%s",
+            AzureEnvironment.AZURE.getResourceManagerEndpoint(),
+            this.manager().subscriptionId(),
+            this.resourceGroupName(),
+            this.name(),
+            deploymentId,
+            this.manager().serviceClient().getApiVersion());
+        HttpRequest request = new HttpRequest(HttpMethod.GET, deploymentStatusUrl);
+        return this.manager().httpPipeline().send(request)
```

Comment:
what the polling do for us?

Last try on 2021-03-01, I rememeber sometimes the API reports "deployment id not found".
If the polling is not for that, we do we need it in the first place? Can we just return the first response without polling?

Reply:
I did not experience 404 error during my tests. I'll see if we can use the first response.

Reply:
From log, "Location" header in `getDeploymentStatus` response is the same as the LRO initial request url. So for now suppose we can repeatedly call `getWithResponse` till we get the final result(with max timeout).

Reply:
But then what is the LRO for? What's the difference of first response and last response?

Reply:
Urls are the same, there's no difference.

Reply:
Then we don't need to polling? Just return first response would be enough?

Reply:
Yes, I think so. First response is enough.